src/cloud-api-adaptor/docs/SecureComms.md

```diff
@@ -0,0 +1,98 @@
+# Secure Communications
+
+Here we describe how to setup the Secure Communications (`Secure Comms` for short) feature of the Adaptor and Forwarder
+
+Secure Comms feature establish an ssh channel by the Adaptor as an ssh client to the Forwarder as an ssh server. The ssh channel is than used for all control communication between the Adaptor and Forwarder. Secure Comms use a two phase approach:
+- Attestation Phase: During this phase an ssh channel is created to enable the Peer Pod to attest against an attestation service. The attestation traffic is tunneled from the Peer Pod to the CAA and from there sent to a KBS as configured at the CAA. Once attestation is achieved, the Forwarder obtains Peer Pod keys from the KBS. The Peer Pod keys are dynamically added to the KBS by the CAA prior to starting the Peer Pod. 
+- Kubernetes Phase: During this phase an ssh channel is created to enable Runtime communicate with teh Kata Agent. This phase uses the keys obtained during the Attestation Phase ensuring that the communication is secured. 
+
+Additional tunnels can be establish via the ssh channel by configuring the Adaptor and Forwarder as described below.
+
+## Setup
+
+### Deploy CAA
+Use any of the option for installing CAA depending on the cloud driver used.
+
+### Deploy KBS-Operator
+Deploy KBS-Operator by following instructions at [https://github.com/confidential-containers/kbs-operator?tab=readme-ov-file#getting-started](KBS Operator Getting Started). 
+
+Copy the kbs-client secret from the `kbs-operator-system` namespace to the `confidential-containers-system` ns. This can be done using:
+
+```sh
+kubectl get secret kbs-client -n kbs-operator-system -o json|jq --arg ns "confidential-containers-system" 'del(.metadata["creationTimestamp","resourceVersion","selfLink","uid","annotations"]) | .metadata.namespace |= $ns' |kubectl apply -f -
+```
+
+For a testing environment, you may need to change the policy of the KBS and AS using the KBS Client to allow all or fit your own policy. One way to do that is:
+
+```sh
+kubectl -n kbs-operator-system exec deployment/kbs-deployment --container as -it -- /bin/bash
+        apt update
+        apt install vim
+        vim /opt/confidential-containers/attestation-service/opa/default.rego // replace default policy to true
+
+kubectl -n kbs-operator-system exec deployment/kbs-deployment --container kbs -it -- /bin/bash
+        apt update
+        apt install vim
+        vim /opa/confidential-containers/kbs/policy.rego // replace default policy to true
+```
+
+
+
+### Build podvm to enforce Secure-Comms
+
+Change the `podvm/files/etc/systemd/system/agent-protocol-forwarder.service` to include:
```

Comment:
@bpradipt - This might be a stupid question, but is there a way that we could use process-user-data to switch the secure comms option rather than having to build a separate peer pod VM for it?

Reply:
@stevenhorsman I think we can move all the agent-protocol-forwarder config to a file and provide it via user-data which process-user-data can read and create the config.
Should be handled as a separate PR imho.